### ISSUE TITLE
Storage.buckets table should be enum 

### ISSUE THREAD

**Issue by spakanati** [nhost/nhost#1002]
Similar to #946 which is about the user roles table, the `storage.buckets` table should be set as enum in Hasura. This would allow for generated types.

**Comment by plmercereau** [comment]
How would we prevent not to introducing a breaking change?

**Comment by elitan** [comment]
I don't think this can be introduced without a breaking change. We should wait to implement this until users can configure their versions with Nhost.

Moving this issue to the Hasura Storage repo.
